erpnext/public/js/utils.js

```diff
@@ -653,4 +653,108 @@ $(document).on('app_ready', function() {
 			});
 		});
 	}
+
+	$.each(["Purchase Receipt", "Stock Entry", "Delivery Note"], function(i, doctype) {
+		frappe.ui.form.on(doctype, {
+			refresh: function(frm) {
+				if(!frm.doc.is_return && frm.doc.status!="Closed") {
+					if (frm.doc.docstatus == 0) {
+						frm.add_custom_button(__('Make Quality Inspection'),
+						function () {
+							let data = [];
+							let inspection_type = " "
+							if (frm.doc.doctype === "Purchase Receipt") {
+								inspection_type = "Incoming";
+							}
+							else if (frm.doc.doctype === "Delivery Note") {
+								inspection_type = "Outgoing";
+							} 
+							else if (frm.doc.doctype === "Stock Entry") {
+								inspection_type = "In Process";
+							}
+							for (let row of cur_frm.doc.items) {
```

Comment:
How are we deciding which items to create QIs for? Isn't there a "Inspection Required" flag for that per item?

Reply:
Im Just adding check for this like fetch from item master and only checked item will come into dialog box